src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java

```diff
@@ -82,22 +109,76 @@ public void generateConditionalOrderPronouncedCoversheetOfflineRespondent(final
                                                               final Applicant applicant,
                                                               final Applicant partner) {
 
+        final Map<String, Object> templateVarsForOfflineRespondent = caseData.isJudicialSeparationCase() && applicant.isRepresented()
+            ? templateVarsForJSSolicitor(caseData, caseId, applicant)
+            : templateVarsForOfflineRespondent(caseData, caseId, applicant, partner);
+
+        final DocumentType coverLetterType = caseData.isJudicialSeparationCase()
+            ? getJSCoverLetterType(applicant.isRepresented(), caseData.isDivorce(), true)
+            : CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2;
+
         caseDataDocumentService.renderDocumentAndUpdateCaseData(
             caseData,
-            CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2,
-            templateVarsForOfflineRespondent(caseData, caseId, applicant, partner),
+            coverLetterType,
+            templateVarsForOfflineRespondent,
             caseId,
-            YES.equals(caseData.getIsJudicialSeparation())
-                ? CO_GRANTED_COVER_LETTER_JS_TEMPLATE_ID
-                : CO_PRONOUNCED_COVER_LETTER_OFFLINE_RESPONDENT_TEMPLATE_ID,
+            getCoverLetterTemplateId(caseData.isJudicialSeparationCase(), applicant.isRepresented(), true),
             applicant.getLanguagePreference(),
-            formatDocumentName(caseId, CONDITIONAL_ORDER_GRANTED_COVERSHEET_DOCUMENT_NAME, now(clock))
+            formatDocumentName(caseId, getCoverLetterDocumentName(caseData, applicant.isRepresented()), now(clock))
         );
     }
 
-    private Map<String, Object> templateVars(final CaseData caseData,
-                                             final Long caseId,
-                                             final Applicant applicant) {
+    private DocumentType getJSCoverLetterType(final boolean isRepresented, final boolean isDivorce, final boolean isOfflineRespondent) {
```

Comment:
I don't believe we need to overwrite the coverletter type. It should remain as the original CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2. Same as we've done for JS flow on Notice of Proceedings. They remained as NOTICE_OF_PROCEEDINGS_APP1 / NOTICE_OF_PROCEEDINGS_APP2, even if solicitor / js / solicitor etc...

Reply:
Yes, but then the document shows as the incorrect type in XUI.  It's not a conditional order document at that point, it's a judicial/separation document.  When caseworkers are going in and out of multiple cases a day, I believe misleading document types (and therefore descriptors) are going to make their jobs more difficult.